pkg/expression/builtin_string.go

```diff
@@ -3770,8 +3774,93 @@ func (b *builtinInstrSig) Clone() builtinFunc {
 	return newSig
 }
 
-// evalInt evals INSTR(str,substr).
+// Find nth occurrence of substring in given string.
+// Search starts from the `position` character to the last character.
+func findNthOccurrence(str, sub string, position, n int) int {
+	currentPos := position
+	for i := 0; i < n; i++ {
+		currentPos = strings.Index(str[currentPos:], sub)
```

Comment:
seems we can add some test when `position` exceed the length of `str`?

Reply:
Added in https://github.com/pingcap/tidb/pull/57890/commits/5d8751f9b18be67d398a5dcd2c96ce6fd2716f02